jina/orchestrate/flow/base.py

```diff
@@ -445,22 +445,33 @@ def _get_pod_addresses(self) -> Dict[str, List[str]]:
 
         return graph_dict
 
-    def _get_k8s_pod_addresses(self, k8s_namespace: str) -> Dict[str, List[str]]:
+    def _get_k8s_pod_addresses(
+        self, k8s_namespace: str, k8s_connection_pool: bool
+    ) -> Dict[str, List[str]]:
         graph_dict = {}
         from jina.serve.networking import K8sGrpcConnectionPool
         from jina.orchestrate.pods.config.helper import to_compatible_name
 
         for node, v in self._pod_nodes.items():
             if node == 'gateway':
                 continue
-            pod_k8s_address = (
-                f'{to_compatible_name(v.head_args.name)}.{k8s_namespace}.svc'
-            )
-            graph_dict[node] = [
-                f'{pod_k8s_address}:{K8sGrpcConnectionPool.K8S_PORT_IN}'
-            ]
 
-        return graph_dict
+            if hasattr(v.args, 'external') and v.args.external:
```

Comment:
A Pod has an external Property that accounts for sandbox as well, better to use that one